network-store-iidm-impl/src/test/java/com/powsybl/network/store/iidm/impl/tck/extensions/MeasurementsTest.java

```diff
@@ -6,114 +6,15 @@
-                .setType(Measurement.Type.REACTIVE_POWER)
-                .setSide(ThreeSides.ONE)
-                .setValue(605.2)
-                .setValid(true)
-                .putProperty("source", "test3")
-                .putProperty("other", "test4")
-                .add();
-
-        Measurements<Load> loadMeasurements = load.getExtension(Measurements.class);
-        assertNotNull(loadMeasurements);
-        assertNull(loadMeasurements.getMeasurement("id"));
-        assertEquals(1, loadMeasurements.getMeasurements().size());
-        Measurement loadMeas = loadMeasurements.getMeasurements().iterator().next();
-        assertNull(loadMeas.getId());
-        assertEquals(Measurement.Type.ACTIVE_POWER, loadMeas.getType());
-        assertNull(loadMeas.getSide());
-        assertEquals(580.0, loadMeas.getValue(), 0.0);
-        assertEquals(5.0, loadMeas.getStandardDeviation(), 0.0);
-        assertFalse(loadMeas.isValid());
-        assertEquals(1, loadMeas.getPropertyNames().size());
-        assertEquals("test", loadMeas.getProperty("source"));
-        assertNull(loadMeas.getProperty("other"));
-
-        loadMeas.remove();
-        assertTrue(loadMeasurements.getMeasurements().isEmpty());
-
-        Measurements<TwoWindingsTransformer> twtMeasurements = twt.getExtension(Measurements.class);
-        assertNotNull(twtMeasurements);
-        assertEquals(2, twtMeasurements.getMeasurements().size());
-        Measurement twtQ1 = twtMeasurements.getMeasurement("MEAS_TWT_Q_2#0");
-        assertNotNull(twtQ1);
-        assertEquals("MEAS_TWT_Q_2#0", twtQ1.getId());
-        assertEquals(Measurement.Type.REACTIVE_POWER, twtQ1.getType());
-        assertEquals(ThreeSides.ONE, twtQ1.getSide());
-        assertEquals(605.2, twtQ1.getValue(), 0.0);
-        assertTrue(Double.isNaN(twtQ1.getStandardDeviation()));
-        assertTrue(twtQ1.isValid());
-        assertEquals(2, twtQ1.getPropertyNames().size());
-        assertEquals("test3", twtQ1.getProperty("source"));
-        assertEquals("test4", twtQ1.getProperty("other"));
-        Measurement twtQ2 = twtMeasurements.getMeasurement("MEAS_TWT_Q_2");
-        assertNotNull(twtQ2);
-        assertEquals("MEAS_TWT_Q_2", twtQ2.getId());
-        assertEquals(Measurement.Type.REACTIVE_POWER, twtQ2.getType());
-        assertEquals(ThreeSides.TWO, twtQ2.getSide());
-        assertEquals(-600.07, twtQ2.getValue(), 0.0);
-        assertEquals(10.2, twtQ2.getStandardDeviation(), 0.0);
-        assertTrue(twtQ2.isValid());
-        assertEquals(1, twtQ2.getPropertyNames().size());
-        assertEquals("test2", twtQ2.getProperty("source"));
-        assertNull(twtQ2.getProperty("other"));
-        twtQ2.putProperty("other", "test5");
-        assertEquals(2, twtQ2.getPropertyNames().size());
-        assertEquals("test5", twtQ2.getProperty("other"));
-
-        twtMeasurements.cleanIfEmpty();
-        assertNotNull(twt.getExtension(Measurements.class));
+    void setValueAndValidity() {
```

Comment:
Just add this line in MeasurementImpl::setValid at the begining : 

MeasurementValidationUtil.checkValue(getValue(), b);

And then you can remove the override of setValueAndValidity ...